### ISSUE TITLE
Remove `additionalProperties: false` from all connector specs

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#14196]
As discovered in https://github.com/airbytehq/oncall/issues/289, a spec declaring `"additionalProperties": false` introduces the risk of accidental breaking changes. Specifically, when removing a property from the spec, existing connector configs will no longer be valid.

We should remove that declaration from all connector specs.

Until this issue is resolved, we should be careful when removing/renaming properties.

**Comment by edgao** [comment]
@grishick fyi, this is followup to https://docs.google.com/document/d/1ru9ApRI7fGtv5gNECchsQe6EB20LlY4UZhJOtEHvOAw/edit#

**Comment by grishick** [comment]
@edgao would it make sense to address this connector by connector starting with GA and Beta connectors?

**Comment by edgao** [comment]
probably, yes. I think the ideal ordering is:
1. Submit a gigantic PR to remove the additionalProperties declaration from ALL connectors spec.json
2. Submit a PR per connector to publish the image, starting with GA/beta, and maybe not even bothering with alpha at all.

Publishing the image is less urgent/important because it only matters if users downgrade the connector. The "remove property + upgrade" case (i.e. what's described in this issue's description) doesn't require a publish.

**Comment by grishick** [comment]
@sherifnada are you OK with the approach @edgao suggested?

**Comment by sherifnada** [comment]
cc @girarda @pedroslopez -- thoughts?

**Comment by girarda** [comment]
yes - it's safe to set additionalProperties to true.

i think this is a great occasion to try publishing multiple connectors at once

**Comment by grishick** [comment]
> yes - it's safe to set additionalProperties to true.
> 
> i think this is a great occasion to try publishing multiple connectors at once

![image](https://user-images.githubusercontent.com/1310940/178075045-90f80986-fe93-474d-9d93-3c219fd9bc6a.png)

**Comment by grishick** [comment]
> yes - it's safe to set additionalProperties to true.
> 
> i think this is a great occasion to try publishing multiple connectors at once

![image](https://user-images.githubusercontent.com/1310940/178075128-ca942321-b62e-4f0f-b80f-9ca329b0151c.png)

**Comment by grishick** [comment]
A search for `additionalProperties": false` in spec.json files in connectors/ folder returns 207 results over 100+ connectors. So, I certainly wouldn't want to make 100+ PRs for this change, but I also would like to avoid blocking progress on prioritized connectors (we have a couple of PRs waiting for this change). To unblock those other PRs, I will turn this into an Epic and create separate issues for unblocking current progress.

**Comment by bazarnov** [comment]
We need to add the exact place in schema where the `additionalProperties` should be `True` but set to false, since now when you debug this, you're blind to see where it should be corrected. Also, we should specify what JSON schema is used (stream schema, spec.json, etc), there is no mention of this. Additionaly, if we have `advancedAuth` block in `spec.json`, should we cover this with `additionalProperties` as well? Please add more explanation from the technical perspective of how this should be properly set up and add more use-cases, not just simple JSON schema with 1 level of nesting. Thanks.

**Comment by edgao** [comment]
@bazarnov could you clarify what you're looking for? This issue is specifically for removing the `additionalProperties: false` from all connectors' `spec.json`, not about adding `additionalProperties: true` anywhere (since by default a schema allows additional properties)

**Comment by bazarnov** [comment]
Thanks for the question, at present, the value of this setting is obvious, no need to dive into a subject.

For all other future and current validation implementations based on checks of specific properties in schemas or input specifications, it would be nice to have the logging message where the issue occurred and what was expected (if applicable to the case).

Giving the general messages like in this test, not always playing good role for debugging purposes.

**Comment by evantahler** [comment]
I think this epic is able to be closed
